pandas/io/formats/format.py

```diff
@@ -1855,29 +1855,16 @@ def just(x):
     return result
 
 
-def _trim_zeros_complex(
-    str_complexes: np.ndarray, decimal: str = ".", na_rep: str = "NaN"
-) -> List[str]:
-    """
-    Separates the real and imaginary parts from the complex number, and
-    executes the _trim_zeros_float method on each of those.
-    """
-    return [
-        "".join(_trim_zeros_float(re.split(r"([j+-])", x), decimal, na_rep))
-        for x in str_complexes
-    ]
-
-
 def _trim_zeros_float(
-    str_floats: Union[np.ndarray, List[str]], decimal: str = ".", na_rep: str = "NaN"
+    str_floats: Union[np.ndarray, List[str]], decimal: str = "."
 ) -> List[str]:
     """
     Trims zeros, leaving just one before the decimal points if need be.
     """
     trimmed = str_floats
 
     def _is_number(x):
-        return x != na_rep and not x.endswith("inf")
+        return re.match(r"\s*-?[0-9]+(\.[0-9]*)?", x) is not None
```

Comment:
is the "\." here assuming `decimal == "."`?

Reply:
Kind of, yeah. I changed to use the decimal argument